packages/orm/src/collections/OneToOneReference.ts

```diff
@@ -57,8 +57,10 @@ export class OneToOneReference<T extends Entity, U extends Entity> extends Abstr
   }
 
   get isSet(): boolean {
-    // This will failure if we're not loaded yet
-    return this.id !== undefined;
+    if (this.isLoaded) {
+      return this.loaded !== undefined;
+    }
+    throw new Error(`${this.entity}.${this.fieldName} was not loaded`);
```

Comment:
question: Why isn't this simply `return this.current() !== undefined`?

I don't think this method is intended to throw an error if it's not set/loaded?

Reply:
This seems close to correct to me.  I think it's just missing a check for a new entity in the if.   A o2o can't know if it's set or not until it's loaded since the FK is on the other entity.  `CustomReference` has the same behavior.

Reply:
@chr1sjf0x there isn't a `current` in this class yet; this is basically an inlined `this.current but error if not loaded`.

> I don't think this method is intended to throw an error if it's not set/loaded?

I think it has to, b/c return `false` in the "not loaded" case would be potentially lying to the caller.

Reply:
Ah, thanks for the explanation @stephenh. I had not considered that the fk is on the `other` side. 👍